packages/bpk-component-dialog/src/common-types.ts

```diff
@@ -28,8 +28,8 @@ export type DialogInnerProps = {
   ariaLabel: string;
   id: string;
   children: ReactNode;
-  dialogRef: (ref: HTMLElement) => void;
-  getApplicationElement: () => HTMLElement;
+  dialogRef?: (ref: HTMLElement) => void;
```

Comment:
having this optional isn't a great solution as it's part of an internal API but exposed publicly.

But without changing all the type structure here there isn't much better option